src/components/Project/Secrets/paginator.js

```diff
@@ -265,31 +282,25 @@ export default class SecretsPaginator extends React.Component {
 
   onClick(idx){   
     const { project } = this.props.data
-    let secret = project.secrets.entries[idx]
-
-    if(secret !== null){
-      // find this
-      this.form = this.initProjectSecretsForm({
-        'key': secret.key,
-        'value': secret.value,
-        'type': secret.type,
-        'id': secret.id,
-        'index': idx,
-        'isSecret': secret.isSecret,
-        'projectID': project.id,
-        'environmentID': this.props.store.app.currentEnvironment.id,
-        'scope': "project",
-      })
-      this.form.$('key').set('disabled', true)
-      this.form.$('isSecret').set('disabled', true)
-
-      this.openDrawer(project)
-    }
+
+    this.setState({selectedSecretID: project.secrets.entries[idx].id})
```

Comment:
Can we modify openDrawer to take the index as a param and then set it inside opendrawer all at once with the other setState?